packages/wdio-browserstack-service/src/Percy/Percy-Handler.ts

```diff
@@ -0,0 +1,176 @@
+            }, 1000)
+        })
+    }
+
+    async percyAutoCapture(eventName: string | null, sessionName: string | null) {
+        try {
+            if (eventName) {
+                if (!sessionName) {
+                    /* Service doesn't wait for handling of browser commands so the below counter is used in teardown method to delay service exit */
+                    this._percyScreenshotCounter += 1
+                }
+
+                (this._browser.percyCaptureMap as PercyCaptureMap).increment(sessionName ? sessionName : (this.sessionName as string), eventName)
+                await (this._isAppAutomate ? PercySDK.screenshotApp((this._browser.percyCaptureMap as PercyCaptureMap).getName( sessionName ? sessionName : (this.sessionName as string), eventName)) : await PercySDK.screenshot(this._browser, (this._browser.percyCaptureMap as PercyCaptureMap).getName( sessionName ? sessionName : (this.sessionName as string), eventName)))
+                this._percyScreenshotCounter -= 1
+            }
+        } catch (err: any) {
+            this._percyScreenshotCounter -= 1;
+            (this._browser.percyCaptureMap as PercyCaptureMap).decrement(sessionName ? sessionName : (this.sessionName as string), eventName as string)
+            PercyLogger.error(`Error while trying to auto capture Percy screenshot ${err}`)
+        }
+    }
+
+    async before () {
+        this._browser.percyCaptureMap = new PercyCaptureMap()
+    }
+
+    deferCapture(sessionName: string, eventName: string | null) {
+        /* Service doesn't wait for handling of browser commands so the below counter is used in teardown method to delay service exit */
+        this._percyScreenshotCounter += 1
+        this.percyDeferredScreenshots.push({ sessionName, eventName })
+    }
+
+    isDOMChangingCommand(args: BeforeCommandArgs): boolean {
+        if ((args.method as string) === 'POST') {
+            if (PERCY_DOM_CHANGING_COMMANDS_ENDPOINTS.includes((args.endpoint as string))) {
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/element') && (args.endpoint as string).includes('click')) {
+                /* click element */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/element') && (args.endpoint as string).includes('clear')) {
+                /* clear element */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/execute') && args.body?.script) {
+                /* execute script sync / async */
+                return true
+            } else if ((args.endpoint as string).includes('/session/:sessionId/touch')) {
+                /* Touch action for Appium */
+                return true
+            }
+        } else if ((args.method as string) === 'DELETE' && (args.endpoint as string) === '/session/:sessionId') {
+            return true
+        }
+        return false
+    }
```

Comment:
Should be the same as:
```ts
```suggestion
    isDOMChangingCommand(args: BeforeCommandArgs): boolean {
        return (
            typeof args.method === 'string' && args.method === 'POST' && typeof args.endpoint === 'string &&
            (
                PERCY_DOM_CHANGING_COMMANDS_ENDPOINTS.includes(args.endpoint) ||
                (
                    args.endpoint.includes('/session/:sessionId/element') &&
                    (args.endpoint.includes('click') || args.endpoint.includes('clear'))
                ) ||
                (
                    args.endpoint.includes('/session/:sessionId/execute') && args.body?.script
                ) ||
                args.endpoint.includes('/session/:sessionId/touch') ||
                (
                    args.method === 'DELETE' && args.endpoint === '/session/:sessionId'
                )
            )
        )
    }
```
I think it would be helpful to add a comment explaining these decision here.

Reply:
Made changes + added comment explaining this handling